assets/training/distillation/src/common/utils.py

```diff
@@ -68,32 +70,39 @@ def newfn(*args, **kwargs):
 
 def get_credential() -> Union[ManagedIdentityCredential, AzureMLOnBehalfOfCredential]:
     """Create and validate credentials."""
-    # try msi, followed by obo, followed by azure cli
-    credential = None
-    try:
-        msi_client_id = os.environ.get("DEFAULT_IDENTITY_CLIENT_ID")
-        credential = ManagedIdentityCredential(client_id=msi_client_id)
-        credential.get_token("https://management.azure.com/.default")
-        logger.info("Using MSI creds")
-        return credential
-    except Exception:
-        logger.warning("MSI auth failed")
+    auth_failures = ""
     try:
+        logger.info("Trying OBO credentials.")
         credential = AzureMLOnBehalfOfCredential()
-        credential.get_token("https://management.azure.com/.default")
-        logger.info("Using OBO creds")
+        credential.get_token(AUTH_TOKEN_SCOPE)
         return credential
-    except Exception:
-        logger.warning("OBO cred failed")
+    except Exception as e:
+        logger.info("Failed to get OBO credentials.")
+        auth_failures += f"OBO: {str(e)}. \n"
+
+    msi_client_id = os.environ.get("DEFAULT_IDENTITY_CLIENT_ID")
+    if msi_client_id:
+        logger.info("Trying ManagedIdentityCredentials.")
+        credential = ManagedIdentityCredential(client_id=msi_client_id)
+        try:
+            credential.get_token(AUTH_TOKEN_SCOPE)
+            return credential
+        except Exception as e:
+            logger.info("Failed to get ManagedIdentityCredential.")
+            auth_failures += f"ManagedIdentity: {str(e)}. \n"
+    
     try:
+        logger.info("Trying AzureCliCredential.")
         credential = AzureCliCredential()
-        credential.get_token("https://management.azure.com/.default")
-        logger.info("Using OBO creds")
+        credential.get_token(AUTH_TOKEN_SCOPE)
         return credential
-    except Exception:
-        logger.error("Azure CLI cred failed")
+    except Exception as e:
+        logger.info("Failed to get AzureCliCredential.")
+        auth_failures += f"AzureCli: {str(e)}. \n"
 
-    raise Exception("Error creating credentials.")
+    auth_failures = "One or more auth methods failed: \n" + auth_failures
+    logger.error(auth_failures)
```

Comment:
Could you please check if `auth_failures` contain any sensitive information?

Reply:
Thanks for the call out @skasturi. They do not contain sensitive information. Individual methods only raise generic exceptions as listed below:

MSI - No managed identity endpoint found.
OBO - AzureML On Behalf of credentials not available in this environment.
AzureCLI - the credential invoked the Azure CLI but didn't receive an access token.